docs/docs/building.md

```diff
@@ -4,158 +4,199 @@ title: Building an upgradeable application
+Next, let's add the implementation of our Basil contract:
 
-```bash
-zos push --deploy-stdlib --network <network>
-```
+    zos add Basil
 
-This will deploy your entire application to the target network, along with the standard library you are using and all its contracts. This way, you can transparently work in development with the contracts provided by the stdlib.
+To link our Basil contract to the OpenZeppelin standard library, we need an Ethereum network where the standard library has already been deployed. But first we want to test this in a local development network, so let's prepare truffle writing this in `truffle.js`:
 
-From there on, you can create proxies for any contract provided by the stdlib:
+    module.exports = {
+      networks: {
+        development: {
+          host: "localhost",
+          port: 9545,
+          network_id: "*"
+        }
+      }
+    };
 
-```bash
-zos create DetailedMintableToken --network <network>
-```
+Then, in a separate terminal, run:
 
+    truffle develop
 
-## Update your smart contract code
+Truffle develop will print 10 accounts. Copy the address of the first one, and then back into the initial terminal, export it as the `OWNER` because it will be useful for us later:
 
-Some time later you might want to change your smart contract code: fix a bug, add a new feature, etc. 
-To do so, update your contracts, making sure you don't change their pre-existing storage structure. This is required
-by **ZeppelinOS** upgradeability mechanism. This means you can add new state variables, but you can't remove the ones you already have. In the example above, this could be the new version of `MyContract`:
+    export OWNER=<address>
 
-```sol
-import "zos-lib/contracts/migrations/Initializable.sol";
+OK, so we finish this step by linking the standard library and pushing our application to the network:
 
-contract MyContract is Initializable {
-  uint256 public x;
-  
-  function initialize(uint256 _x) isInitializer public {
-    x = _x;
-  }
-  
-  function y() public pure returns (uint256) {
-    return 1337;
-  }
-}
-```
+    zos link openzeppelin-zos
+    zos push --network development --deploy-stdlib
 
-Use truffle to compile the new version of your code:
-```sh
-npx truffle compile
-```
+We pass that `--deploy-stdlib` flag because we are using a development network that started clean. When you deploy your application to a real network where the `openzeppelin-zos` standard library has already been deployed, you won't need this flag.
 
-We'll now use `zos` to register and deploy the new code for `MyContract` to the blockchain. Sync the new version of your project by running: 
+TODO: test this and comment about the resulting logs. On the published zos, the set-stdlib command fails, wait for a new release.
```

Comment:
I wouldn't leave a TODO here, can you create an issue on this? Also, be sure to link it from: https://github.com/zeppelinos/labs/issues/44.

Reply:
I noticed I later added a comment here. I just removed this TODO, will check later if it needs to be expanded and if so, file a bug.